controller/share_manager_controller.go

```diff
@@ -1271,6 +1274,9 @@ func (c *ShareManagerController) createShareManagerPod(sm *longhorn.ShareManager
 			}
 			tolerationsFromStorageClass := c.getShareManagerTolerationsFromStorageClass(sc)
 			tolerations = append(tolerations, tolerationsFromStorageClass...)
+
+			// A storage class can override mkfs parameters which need to be passed to the share manager
+			formatOptions = c.splitFormatOptions(sc)
```

Comment:
Why not move the parsing function to longhorn-share-manager? We can simply assign `sc.Parameters["mkfsParams"]` to the env `FS_FORMAT_OPTIONS`?

Reply:
>  I see your point though - why split, join and then split again?

Yes, you are right.

Thanks for the explanation. I'm fine with the current implementation.
Let me know if the PRs are ready for review. Thank you.

Reply:
If I should change it thats fine for me. Otherwise PRs are ready for review now.
I'll fill out the ready-for-testing checklist in parallel